### ISSUE TITLE
Enable Customer Usage Attribution for partners

### ISSUE THREAD

**Issue by richeney** [Azure/ALZ-Bicep#265]
The CUA mechanism is being used to allow tracking of the various modules and for Microsoft to provide better support.

However, [Customer Usage Attribution](https://aka.ms/customerusageattribution) is one of the mechanisms used to recognise partner attached ACR.  It isn't the preferred mechanism - CSP and PAL are more robust - but is still valid and the repo should enable its use.

I suggest that the repo introduces a consistently named parameter, e.g. `CUA_GUID`, to all modules so that a partner specified GUID can filter through the modules. The parameter value would be set to the [GUID generated and registered](https://docs.microsoft.com/azure/marketplace/azure-partner-customer-usage-attribution#create-guids) by the partner.

**Comment by jtracey93** [comment]
Hey @richeney,

Thanks for raising this. This needs some consideration as it would technically break our own telemetry if we allowed the CUA ID to be set via a parameter, today its a variable, that partners could change if they wanted to today once they've forked/cloned the repo.

I agree that CSP and PAL are the better ways for partner to attach themselves to ACR.

Are there any active requests for this from partners (feel free to ping me on Teams to keep it confidential 👍)? This will help us understand priority for this work.

### Potential Idea

Make the telemetry module a loop that merges the `ALZ-Bicep` CUA ID with an input parameter version in a concat and then loop over the combined array.

That way both PIID deployments could occur and live together.

Would you want the opt out of telemetry option to apply to the partner one also @richeney?

**Comment by richeney** [comment]
To be clear, CUA is a mechanism intended for use by partners. 

I don't think it supports multiple GUIDs as suggested by the loop idea. It just adds pid-GUID to the agent header and then uses the correlationId to determine which resources should be linked to the MPN ID via that pre-registered GUID.

**Comment by jtracey93** [comment]
**Update:**

Myself and @richeney are syncing internally on this

**Comment by jtracey93** [comment]
@richeney did we need to pick this one backup?

**Comment by richeney** [comment]
Yep, I think support for [Customer Usage Attribution](https://aka.ms/customerusageattribution) needs to be added. I understand that CUA is being used to track impact, but it is there as a partner attach ACR mechanism and partners should be able to use it here. That is analogous to the azurerm Terraform provider where Hashicorp use it by default to get some visibility on usage, but it is always [available to partner](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#partner_id) (or to be [switched off](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#disable_terraform_partner_id)) even if that has an impact on what Hashicorp can see.

Note that there is a new partner program and partner score for this FY which has added focus on partner ACR attach, so we have an opportunity here to add support and avoid any unnecessary friction.

I suggest CUA in Bicep need to support both of the following formats:

* `pid-<guid>-partnercenter` ([Marketplace](https://docs.microsoft.com/en-gb/azure/marketplace/azure-partner-customer-usage-attribution#commercial-marketplace-azure-apps) - this is the format generated in Partner Center after publishing offers)
* `pid-<guid>` ([Other use cases](https://docs.microsoft.com/en-gb/azure/marketplace/azure-partner-customer-usage-attribution#other-use-cases) - original format, created manually and registered)


For reference, I had a recent PR in the Terraform azurerm provider which was pulled into the v3.21.0 release. [Issue 17403](https://github.com/hashicorp/terraform-provider-azurerm/issues/17403) / [Pull 17441]( https://github.com/hashicorp/terraform-provider-azurerm/pull/17441).

Adding @mormond, who's the Marketplace guru in the team!

**Comment by jtracey93** [comment]
Ado sync
